TFUtil.py

```diff
@@ -8085,6 +8105,15 @@ def tensor_array_element_shape(ta):
   return tf.TensorShape(None)
 
 
+def tensor_array_element_shape_py3_tf_1_14_and_later(ta):
+  """
+  :param tf.TensorArray ta:
+  :rtype: tf.TensorShape
+  """
+  assert isinstance(ta.element_shape, tf.TensorShape)
+  return ta.element_shape
```

Comment:
Do not create an extra function for this. Just make it inline in `tensor_array_element_shape`. The same also for `tensor_array_element_shape_tf_up_to_1_13`.